---
fig-cap-location: top
---

# **Day 1 - Learning objectives**

Today, we will get introduced to the first steps of analysing RNA sequencing data using the command line - which includes how to pre-process raw sequencing reads to analysis ready count data.
The main topics of interest are- 

### **RNA-Seq**
- What is RNA-Seq?
- What are the main applications of RNA-Seq
- How does a basic RNA-seq workflow look like?
- What is the experimental case-study for this workshop?

### **A specific case-study using RNA-Seq**
- Knockout mouse model to study Williams-Beuren Syndrome (WBS)
- What is the Experimental question?
- What does the experiment design look like?

### **Understand the steps involved in differential expression analysis using bulk RNAseq data**
- Check sequencing read quality
- Trimming seqeuncing reads 
- Alignment against the reference genome
- Generate the gene-count matrix
- Differential expression analysis (Day2)
- Functional interpretation        (Day2)

### **Describe the benefits of using nf-core workflows**
- A typical bioinformatics workflow.
- Nextflow as a workflow management systems, how does it help?
- Use nf-core’s rnaseq pipeline to transform raw sequencing data into analysis ready count data.

### **Deploy the nf-core workflow on Pawsey’s Nimbus VM**
- What is the core nf-core-rnaseq command?
- What are the main inputs?
- What are the main outputs? 
- Can we add/remove/change the tools/parameters used for the various processes?
- What do we carry forward to DAY 2: The gene-count matrix

___
**Notes**   
<sup id="f1">1[↩](#a1)</sup> Here is a list of ['nf-core pipelines'](https://nf-co.re/pipelines/).

___




<div class="keypoints">

### Key points

- We will use the nextflow `nf-core/rnaseq` pipeline to convert raw reads to gene-counts.
- We will run the `nf-core/rnaseq` pipeline on a `Pawsey’s Nimbus VM instance`.

</div>  
