---
fig-cap-location: top
---

# **Day 1 - Learning objectives**

Today, we will introduce you to the first steps of analysing RNA sequencing (RNA-seq) data using the command line which includes how to pre-process raw sequencing reads to analysis-ready count data.

### **Introductory slides**
- Understand how RNA-seq data is generated
- Review the applications of RNA-seq
- Understand experimental design considerations for differential expression (DE) analysis
- Understand the basic RNA-seq DE analysis workflow


### **Run nf-core/rnaseq pipeline on Pawsey Nimbus**
- Set up your computer for this workshop series
- Log in to your Nimbus instance
- Downloading files to Pawsey Nimbus
- Run the nf-core/rnaseq command to excecute the pipeline 


### **Why use nf-core workflows?**
- Understand why Nextflow and nf-core are good options for reproducible and portable bioinformatics workflows

### **The nf-core-rnaseq command**
- What is the core nf-core-rnaseq command?
- What are the main inputs?
- What are the main outputs? 
- Can we add/remove/change the tools/parameters used for the various processes?
- What do we carry forward to DAY 2: The gene-count matrix


### **Understand the steps involved in differential expression analysis using bulk RNAseq data**
- Check sequencing read quality
- Trimming seqeuncing reads 
- Alignment against the reference genome
- Generate the gene-count matrix
- Differential expression analysis (Day2)
- Functional interpretation        (Day2)

___
**Notes**   
<sup id="f1">1[↩](#a1)</sup> Here is a list of ['nf-core pipelines'](https://nf-co.re/pipelines/).

___




<div class="keypoints">

### Key points

- We will use the nextflow `nf-core/rnaseq` pipeline to convert raw reads to gene-counts.
- We will run the `nf-core/rnaseq` pipeline on a `Pawsey’s Nimbus VM instance`.

</div>  
