---
fig-cap-location: top
---

# **Downloading required files**

<div class="questions">

### **Questions**

- How do I setup my base directory?
- How do I download the required input files on Nimbus?

</div>  

<div class="objectives">

### **Objectives**
Download input data to an appropriate path
</div>  

    
### **Open your terminal**
Go back to the terminal window in which you have logged on to Nimbus . If you have accidently closed it, you will need to repeat steps in [Setup]() again.



::: {.callout-note appearance="simple"}
## Copying the code from the `grey` boxes on training materials
In this workshop we need to copy code from the grey boxes in the training materials and run it in your terminal
e.g 
```default
ssh training@###.###.###.###
```
:::


#### Make a directory called 'base_directory'
```default
mkdir base_directory
```

#### Download a compressed file (.tar.gz) to set up the working directory

In your terminal, type the following to move into to the base_directory:
```default
cd base_directory
```
Then type the following to download the input files:
```default
git clone https://github.com/Sydney-Informatics-Hub/RNASeq_materials.git
```

The files are packaged up to save space. Move the file to the current directory and Unpack it:
```default
mv RNASeq_materials/working_directory.tar.gz ./
tar -zxvf working_directory.tar.gz
```

This creates a directory 'working_directory'. Move into this directory using the command
```default
cd working_directory
```

The `working_directory` folder contains the following files
```default
- 6 fastq files : SRR*_selected.fastq.gz
- A samplesheet file with details about the fastq files
- A nextflow.config file
- FULL_count_matrix.txt
- rnaseq_DE_Full_matrix_DryRun.Rmd (Required for Day 2)
- rnaseq_DE_Full_matrix_DryRun.html (Required for Day 2)
```

You can list all the files in your current directory by typing `ls -lh`. 

<font size="3">
```default
-rw-------@ 1 deshpandes  staff   919K 22 Apr 15:11 FULL_count_matrix.txt
-rw-------@ 1 deshpandes  staff   1.3K 30 Aug 11:53 README
-rwxr-xr-x@ 1 deshpandes  staff   3.4M 28 Jul 06:11 SRR3473984_selected.fastq.gz
-rwxr-xr-x@ 1 deshpandes  staff   3.1M 28 Jul 06:11 SRR3473985_selected.fastq.gz
-rwxr-xr-x@ 1 deshpandes  staff   3.0M 28 Jul 06:11 SRR3473986_selected.fastq.gz
-rwxr-xr-x@ 1 deshpandes  staff   3.3M 28 Jul 06:11 SRR3473987_selected.fastq.gz
-rwxr-xr-x@ 1 deshpandes  staff   4.2M 28 Jul 06:11 SRR3473988_selected.fastq.gz
-rwxr-xr-x@ 1 deshpandes  staff   3.0M 28 Jul 06:11 SRR3473989_selected.fastq.gz
-rw-------@ 1 deshpandes  staff   4.8K 28 Jul 06:11 nextflow.config
-rw-r--r--@ 1 deshpandes  staff    18K 30 Aug 11:33 rnaseq_DE_Full_matrix_DryRun.Rmd
-rw-r--r--@ 1 deshpandes  staff   5.0M 30 Aug 11:37 rnaseq_DE_Full_matrix_DryRun.html
-rw-------@ 1 deshpandes  staff   330B 28 Jul 07:25 samplesheet.csv
```
</font>


### **Samplesheet sequence data**
- We need a [samplesheet](https://nf-co.re/rnaseq/3.7/usage#samplesheet-input) with information about the samples you would like to analyse.
- If the samplesheet is located in a different folder, you can use this parameter to specify its path. 

![](/fig/elaborate_samplesheet.png)

- The samplesheet is a comma-separated file with 4 columns, and a header row as shown in the example samplesheet above. 
- The pipeline will auto-detect whether a sample is single- or paired-end using the information provided in the samplesheet. 
- The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 4 columns to match those defined in the table below.

![Samplesheet columns](/fig/samplesheet_description.png)



<div class="challenge">

### Challenge
- The samplesheet which we will be using for today's hands-on is already created and available at `~/base_directory/working_directory/samplesheet.csv`.
- Can you open this file using the utility `vi` and figure out what is the difference between ours and the one shown as a example, above?

<details>
<summary>Solution</summary>
- The above samplesheet consists of both single- and paired-end data. It is for 6 samples, where TREATMENT_REP3 has been sequenced twice.
- The samplesheet which we are using in today's analysis is a lot simpler, with only single-end reads;  Note that the place-holders for the reverse-reads is empty. 

![Today's samplesheet](/fig/our_samplesheet_final.png){width=70%}

</details>
</div>  



Check your path by typing:
```default
pwd
```
- You should be in the path: ```/home/training/base_directory/working_directory``` 
- If you are not, move into the above path by typing:
```default
cd /home/training/base_directory/working_directory
```


- Keep this command-line (terminal) window open for further analysis. 
- We can proceed to the section `Differential expression(DE) analysis` by clicking on the top menu-bar.





<div class="keypoints">

### **Key points**
All the required files for this workshop are downloaded as a single compressed file
</div>  

