---
fig-cap-location: top
from: markdown+emoji
---

# **Run the nf-core/rnaseq pipeline**

<div class="questions">

### **Questions** {.unlisted}
* How to run the nextflow command in terminal?
</div>
|
|
Now that we have prepared our working space and downloaded all the data we need, we can run the nf-core/rnaseq pipeline. The mouse genome is large and contains ~30,000 genes which would take hours to process. For the sake of expediency, today we are running the pre-processing workflow on a subset of the whole mouse genome. Tomorrow we will provide you with a whole genome count matrix, to perform differential expression and enrichment analyses. 

::: {.callout-note appearance="simple"}

#### Why run the command now?
We need to allow ~20 mins for the command to run on the test data. For this reason we will run the command first and then discuss each step of this pipeline once it has completed. 

:::

## **Execute the nf-core/rnaseq run command**

In your terminal, check your path by running:
```default
pwd
```

It should return: 

```default
/home/training/base_directory/working_directory
```

If you are not, move into the above path by running:

```default
cd /home/training/base_directory/working_directory
```

In order to run the pre-processing pipeline for all 6 samples, we need to provide nf-core/rnaseq command with some essential parameters and inputs. Our run command specifies some extra details that we will explore in more depth later today, but for now the important things to know are:

* All the reference data we are working with (i.e. fasta, gtf, star index files) are stored in a centralised file system called CernVM File System (CVMFS) (`cvmfs_path=`) that we are accessing remotely. 
* We have specified how much memory (`--max_memory`) and how many CPUs (`--max_cpus`) the nf-core/rnaseq pipeline can use. Doing this ensures the pipeline won't try to consume more memory or CPUs than are available on our system and crash. 
* Outputs will be saved to the results directory (`--outdir`)

Run the command:  

<font size="2.7">
```default
cvmfs_path=/cvmfs/data.biocommons.aarnet.edu.au/Final_resources_250722

nextflow run $cvmfs_path/nfcore_pipeline/rnaseq/ \
                --input samplesheet.csv \
                -profile singularity \
                --fasta $cvmfs_path/Mouse_chr18_reference/chr18.fa \
                --gtf $cvmfs_path/Mouse_chr18_reference/chr_18_startOfLine.gtf \
                --star_index $cvmfs_path/Mouse_chr18_reference/chr18_STAR_singularity_index/ \
                --max_memory '6 GB' --max_cpus 2 \
                --outdir results \
                -with-report excecution_report.html \
                -with-timeline timeline_report.html

```
</font>

The progress of the workflow will be displayed in the terminal and updated in real-time. Take a look at all the processes you have just run with a single command! 

::: {.callout-note appearance="simple"}

#### What if my command fails?
Don't worry if your run fails for whatever reason, we have a set of results available for anyone who has experiences issues. If you need help with this, please ask one of our demonstrators in the Slack channel. 

:::

If you are an experienced CLI user, you might want to run commands that take a while to finish in the background, so you can keep using your terminal while it runs. If that sounds nifty to you, feel free to try the challenge below to run your nf-core/rnaseq command. 

<div class="challenge">

### **Challenge for experienced CLI users** {.unlisted}
We can run nf-core/rnaseq command using a utility called __screen__. The __screen command__ provides the ability to launch and use multiple shell sessions from a single ssh session. Can you:

1. Create a screen session  
2. Run the nf-core/rnaseq run command above inside that session  
3. Detatch from the screen session 
4. Reopen the screen session to check on the progress of the run command   

Hint: to work out what flags you will need to do this, run: 

```default
screen --help 
```

<details>
<summary>Solution</summary>

1. To create a new screen session with the name run_nextflow_in_screen, run: 
```r
screen -S run_nextflow_in_screen
```

3. To detatch the screen session, but keep it running in the background, run:

```default
screen -d
``` 
This can also be done with the help of shortcut key ```Ctrl-a + d```

4. To reattach the screen session, run: 

```default
screen -r run_nextflow_in_screen
```

</details>
</div>  

For now, we will leave the pipeline to run in our terminals and discuss how Nextflow and nf-core/rnaseq work while we wait. Keep your terminal window open to allow the nf-core/rnaseq command to finish and proceed to the next lesson: [Why use nf-core/rnaseq? > Introduction to Nextflow](). **GS- add links here once finalised** 

<div class="keypoints">

### **Key points** {.unlisted}

* The nf-core/rnaseq pipeline is a simplified, easy to run solution for RNA-seq data pre-processing

</div>  



  
