---
fig-cap-location: top
---

# **A typical differential expression (DE) analysis pipeline**

<div class="questions">

### **Questions**

- What are the different steps in a typical RNA-seq differential expression (DE) pipeline?
</div>  

<div class="objectives">

### **Objectives**

- Get familier with the RNA-seq DE analysis pipeline.

</div>  

<div class="challenge">

### **Challenge**
- Let's try and ask ourselves a question, how should a typlical RNA-seq differential expression (DE) pipeline look like?
- To assist, we have provided the 5 basic processes as building blocks in the figure below. **But** these are placed in no particular order.
- Can you try and arrange them sequentially from Step (1) to Step (5). 

![Arrange the processes from the differential expression analysis in a sequential order](/fig/Basic_DE_pipeline_puzzle.png)

<details>
<summary>Solution</summary>

![A typical differential expression analysis pipeline](/fig/Basic_DE_pipeline.png){width=50%}

</details>
</div>  



<br>So the basic steps in a differential expression (DE) pipeline are:

### 1. Raw sequence data and quality check
- Obtain sequencing data in [Fastq](https://sapac.support.illumina.com/bulletins/2016/04/fastq-files-explained.html) format 
- Quality checking with [fastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and [multiQC](https://multiqc.info/examples/wgs/multiqc_report.html)
- Quality trimming with [Trim galore](bioinformatics.babraham.ac.uk/projects/trim_galore/)

### 2. Alignment and Visualisation
- Alignment with [STAR](https://github.com/alexdobin/STAR) aligner
- Visualisation using [samtools](http://www.htslib.org/) /  [IGV](https://software.broadinstitute.org/software/igv/home)

### 3. Mapped reads to gene-counts
Converting the reads mapped to the reference genome to a matrix which contains individual gene-counts per sample for downstream statistical enrichment analysis. 

### 4. Differential expression (DE) analysis
Identify statistically significant DE genes using tools such as the R-library [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html).

### 5. Functional interpretation
Functional enrichment analysis using tools such as the R-library [clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html). 


We will discuss each of the steps from (1) to (3) in detail today. The steps (5) and (6) will be covered on Day 2 tomorrow.


<div class="keypoints">

### **Key points**

- Tools such as `FastQC` can be used for assessing the sequencing read-quality.
- `Trim galore` is one of the many tools available for read-trimming based on multiple criteria.
- We will use the aligner `STAR` to map the reads to the reference genome/transcriptome.
- The identification of statistically significant DE genes can be done using R libraries such as `DeSeq2`.

</div>  






