---
fig-cap-location: top
---

# **Raw sequence data and quality check**


<div class="questions">

### **Questions**

- What is a FastQ format?
- What are Phred quality scores?
- How to evaluate the quality using fastQC?
</div>  

<div class="objectives">

### **Objectives**

- Understand FASTQ format.
- Understand the quality metrix of FASTQ using fastQC.
</div>  



### **FastQ format**

![](/fig/fastq_format.jpg)

- [FASTQ](https://en.wikipedia.org/wiki/FASTQ_format) format is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores.
- Both the sequence letter and quality score are each encoded with a single ASCII character for brevity.


<div class="challenge">

### **Challenge #1**
- How many sequences were in SRR3473989_selected.fastq? 
<br>**NOTE:** This is one of the fastq files which we have downloaded. 
- What are the lengths of the reads in SRR3473989_selected.fastq?


<details>
<summary>Solution</summary>
- There are `104,404` sequences in the file SRR3473989_selected.fastq.
- The reads in the file SRR3473989_selected.fastq are of length `100 bp`.

</details>
</div>  



### **Phred Quality Scores**

![](/fig/Phred_short.png)

- The quality score which assess reliability of each base call is calculated as Phred quality score.
- Several different encodings have been used throughout the years to report on this by Illumina, and the [Wikipedia](https://en.wikipedia.org/wiki/FASTQ_format) page is the best place to figure out the most updated.


### **FastQC**

![](/fig/fastqQC.png)

- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines.
- It provides a modular set of analyses which you can use to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis.



<div class="challenge">

### **Challenge #2**
- Looking at the `Per base sequence quality plot` in the above figure, 
    - Which part of the reads tend to have worse per base sequence quality? 
    - Do you think this dataset is of “good” quality? 
    - Any suggestions to make it better?

<details>
<summary>Solution</summary>
- Reads which tend to have worse per base sequence quality are towards the right hand side (3'-end).
- The color coding separates out regions of good quality (Red Q>28) from the rest. Overall yes, as most of the regions of the reads show quality values in `red`.
- We can trim the bases towards the 3'-end and hope to improve the overall read-quality. But trimming by quality for RNA-seq data has its pros and cons.

</details>
</div>  




<div class="challenge">

### **Additional Challenge**
- Using corresponding ‘Dec’ values in this ASCII table (http://www.asciitable.com/ ), what is the Phred quality score of the first base in read provided in SRR3473984_selected.fastq?
- Now that you have determined the Phred score for the first base in the previous example, what is the probability (P) that this base was incorrectly called by the sequencing machine?
- Why do we use ASCII to encode quality scores?
    <br>**Please try this challenge in your free time**

<details>
<summary>Solution</summary>
My solution is here

</details>
</div>  




<div class="keypoints">

### Key points
- `FastQ` is a standard format for sequencing reads.
- `Phred Quality Scores` are a useful metrix to check the read-quality. Tools such as `FastQC` can be used for avaluation. 

</div>  

