---
fig-cap-location: top
---

# **Customising the nf-core/rnaseq workflow**


<div class="questions">

### **Questions**{.unlisted}

- What are the essential nf-core/rnaseq parameters?
- What parameters can be customised? 
</div>  
|
|
The nf-core/rnaseq pipeline can be run using a single command with deafult parameters. The default parameters are explained in detail here, and include: 

* Either an input sample sheet (`--input`) or directory housing fastq files (`--reads`)  
* A reference genome, either one available through Illumina's [iGenomes repository](https://github.com/nf-core/rnaseq/blob/master/conf/igenomes.config) (`--genome`), or a user-specified reference assembly (`--fasta`) and annotation file (`--gtf`) 
* A configuration profile suitable for the compute environment you're working on. 

For example, a user working with human samples, who has Singularity installed and all fastq files stored in a directory would be able to run this command to run the nf-core/rnaseq pipeline: 

<font size="3">
```default
nextflow run nf-core/rnaseq \                 
  --reads '*_R{1,2}.fastq.gz' \               # The location of fastq files
  --genome GRCh38 \                           # Illumina iGenomes database
  -profile singularity                        # For pre-installed software
```
</font>

Most of us will need to customise the command a little more than this though. For example, a user working with multiple samples, who wants to provide their own pre-indexed reference data, and has compute resource limitations might run a command that looks more like this: 


<font size="3">
```default
nextflow run $Path_to_nf-core/rnaseq \      
  --input $Path_to_samplesheet.csv \        # Samplesheet file-name
  -profile singularity \                    # For pre-installed software
  --fasta $Genome_fasta_file \              # Genome sequence file
  --gtf $Path_to_Genome.gtf \               # GTF - gene locations on genome
  --star_index $Path_to_index file \        # Formatted file for aligner
  --max_memory '6 GB' --max_cpus 2 \        # Memory and cpu resources 
  --outdir $Path_to_results \               # Results folder
  -with-report excecution_report.html \     # Excecution log file-name 
  -with-timeline timeline_report.html       # Timeline log file-name
```
</font>


### **Some useful customisation options**

#### *Input and output options*
```default
--input   Path to comma-separated file containing information about the samples.
--outdir  The output directory where the results will be saved.
```

#### *Reference genome options*

```default
--genome            Name of iGenomes reference. 
--star_index        Path to directory or tar.gz archive for pre-built STAR index.
--hisat2_index      Path to directory or tar.gz archive for pre-built HISAT2 index.
--save_reference    If generated by the pipeline save the STAR index in the results directory.
```

#### *Alignment options*
```default
--aligner           Alignment algorithm to use.
--pseudo_aligner    Pseudo aligner to use.
```

#### *Process skipping/use-alternate options*
```default
--deseq2_vst      Use vst transformation instead of rlog with DESeq2.
--skip_fastqc     Skip FastQC.
--skip_multiqc    Skip MultiQC.
```

For details of all parameters in nf-core take a look [here](https://nf-co.re/rnaseq/3.8.1/parameters).

<div class="keypoints">

### **Key points**{.unlisted}

- A single nf-core command can be run for the complete pipeline. 
- Many different parameters can be used to customise pipeline runs. 
- nf-core allows users to skip non-madatory steps in the pipeline.

</div>  
