---
fig-cap-location: top
---

# **Raw sequence data and quality check**


<div class="questions">

### **Questions**

- What is a FastQ format?
- What are Phred quality scores?
- How to evaluate the quality using fastQC?
</div>  

<div class="objectives">

### **Objectives**

- Understand FASTQ format.
- Understand the quality metrix of FASTQ using fastQC.
</div>  

![**Stage 1 of `nf-core/rnaseq`: (Pre-processing-fastQC):**](../fig/nfcore_stage1.1.png)

#### **We have previously seen the `FastQ` format**
[FASTQ](https://en.wikipedia.org/wiki/FASTQ_format) format is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores.
![](/fig/fastq_format.jpg)
<br>Both the sequence letter and quality score are each encoded with a single ASCII character for brevity.

#### **What are Phred quality scores?**
The quality score which assess reliability of each base call is calculated as Phred quality score.<br>
![](/fig/Phred_short.png)
<br>Several different encodings have been used throughout the years to report on this by Illumina, and the [Wikipedia](https://en.wikipedia.org/wiki/FASTQ_format) page is the best place to figure out the most updated.


#### **How can you evaluate quality using the tool `FastQC`**

![](/fig/fastqQC.png)

- Quality checking with [fastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) and [multiQC](https://multiqc.info/examples/wgs/multiqc_report.html)


- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines.
- It provides a modular set of analyses which you can use to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis.


<div class="challenge">
### **Challenge #1**
- How many sequences were in SRR3473989_selected.fastq? 
<br>**NOTE:**This is one of the fastq files which we have downloaded.
- What are the lengths of the reads in SRR3473989_selected.fastq?

<br>**HINT** fastqQC log files can help.

<details>
<summary>Solution</summary>
- There are `59887` sequences in the file SRR3473989_selected.fastq.
- The reads in the file SRR3473989_selected.fastq are of length `101 bp`.

</details>
</div>  


<div class="challenge">

### **Challenge #2**
- Looking at the `Per base sequence quality plot` in the above figure, 
    - Which part of the reads tend to have worse per base sequence quality? 
    - Do you think this dataset is of “good” quality? 
    - Any suggestions to make it better?

<details>
<summary>Solution</summary>
- Reads which tend to have worse per base sequence quality are towards the right hand side (3'-end).
- The color coding separates out regions of good quality (Red Q>28) from the rest. Overall yes, as most of the regions of the reads show quality values in `red`.
- We can trim the bases towards the 3'-end and hope to improve the overall read-quality. But trimming by quality for RNA-seq data has its pros and cons.

</details>
</div>  


<div class="keypoints">

### Key points
- `FastQ` is a standard format for sequencing reads.
- `Phred Quality Scores` are a useful metrix to check the read-quality. Tools such as `FastQC` can be used for avaluation. 

</div>  

