---
fig-cap-location: top
---

# Trimming


<div class="questions">

### Questions

- Why trim RNA sequencing reads?
</div>  

<div class="objectives">

### Objectives

- Understand the requirement for trimming RNA sequencing reads
- Understand the pros and cons of trimming

</div>  


- Trimming is sometimes performed to improve the quality of the raw data and potentially improve its mappability. There are several ways to perform trimming:
  - Removal of poor quality reads or bases (e.g. ends of reads)
  - Removal of adapter sequences
  - Removal of polyA tails

__NOTE:__
- Be very wary about trimming RNA sequencing data
  - Trimming of poor quality reads can affect gene expression estimates (Williams at el., 2016)

- Trimming of high quality adapter sequences was shown to increase quality and reliability of biological signals in RNA-seq data (Dozmorov et al., 2015)


<div class="challenge">

### Challenge
```r
From looking into the results folder in the path `~/base_directory/working_directory/`, can you tell :
- Which tool does nfcore-rnaseq use for read-trimming?
- Was there an effect of the trimming step on the sample SRR3473989_selected.fastq?
  - Which tool did you use for visualisation of the quality; Before and after trimming?
  - Which folders in the 'results' did you need to access? 
```

<details>
<summary>Solution</summary>



</details>
</div>  





<div class="keypoints">

### Key points

- Tools such as Trim galore and trimmomatic can be used for read-trimming
- FastQC can be used to visualise if the trimming step helped improve the read-set.

</div>  



  

