---
fig-cap-location: top
---

# Alignment to reference genome"

<div class="questions">

### Questions

- How to map seqeuncig reads to a reference genome?
</div>  

<div class="objectives">

### Objectives

- Understand the alignment/mapping process
- Understand the outout sam/bam formats

</div>  

![](/fig/alignment_counting_nfcore-rnaseq.png)

- After trimming, the next step of processing RNA-seq data is to map the reads to a reference genome. 
- RNA-seq specific aligner needs to be used to accomodate splicing.
- The most commonly used aligners for short-read sequencing are STAR, HISAT2 etc.
- The output of the alignment step is a [bam](https://en.wikipedia.org/wiki/SAM_(file_format)) file, which is a compressed ("B"inary) form of a [sam](https://en.wikipedia.org/wiki/SAM_(file_format)) file.






<div class="keypoints">

### Key points

- Nimbus VM instances from Pawsey provide on-demand access for scalability.
- We use a ssh client to connect to a Linux instance.

</div>  

