---
fig-cap-location: top
---

# **Read alignment and quantification**

<div class="questions">

### Questions{.unlisted}

- How to map sequencing reads to a reference genome?
- How to use the mapped alignments to quantify gene-counts
</div>  

We've moved on to the second stage of the nf-core/rnaseq workflow: read alignment and quantification (red box below). 

![](../fig/nfcore_stage2And3.png)

### **Alignment to reference genome**

**GS- can you confirm this is correct please.**  
After read trimming, the nf-core/rnaseq pipeline maps reads to a reference genome specified by the user. Keep in mind that RNA-seq specific alignment tools are different to whole genome alignment tools. They are designed to be **'splice aware'**, meaning they are capable of differentiating intronic from exonic regions in the alignment process. 

Bulk RNA-seq reads are derived from mature mRNA and contain only exons (and no introns). This means a sequence read can span 2 exons. Splice-aware aligners use a reference **genome**, not a transcriptome to perform read alignment, which contain intergenic, intronic and exonic sequences. This means that when they're aligned to the reference genome, RNA-seq reads might span large introns. Splice aware aligners know not to align the RNA-seq reads to introns and can align a read across exons. 

::: {.callout-note appearance="simple"}

#### Genetics 101 refresher: mRNA splicing
mRNA splicing is the process by which an mRNA transcript prepares to be translated into an amino acid sequence. It works by removing introns and splicing all exons back together to create a mature mRNA that is transported from the nucleus to the cytoplasm, ready to undergo translation. 

:::

The most commonly used aligners for short-read sequencing are available in the nf-core/rnaseq workflow, including [STAR](https://github.com/alexdobin/STAR) and [HISAT2](http://daehwankimlab.github.io/hisat2/). Aligned sequences for each sample are output in the [bam](https://en.wikipedia.org/wiki/BAM_(file_format)) file format.  

The challenge activity for this session will be a group exercise. In breakout rooms, your facilitator will demonstrate how we can visualise alignments using a tool called [IGV](https://software.broadinstitute.org/software/igv/). 

**GS- the link to the case study is broken**

<div class="challenge">

### **Challenge**{.unlisted}
This is a hard one! Thinking back to the [case study](), can you work out how you would handle the scenario below? 

Imagine that **we have forgotten to label the conditional groupings of our samples and we don’t know which samples belong to the wildtype or knockout groups!** 

Can we use the alignment files to assign samples to their correct treatment group? 

<details>
<summary>Solution</summary>
- From the case study, we know a loss of function mutation of *Gtf2ird1* was generated by an insertion of a Myc transgene, resulting in a 40 kb deletion surrounding exon 1.
- Navigate to the gene in the mm10 assemble (chr5:134332897-134481480)
- Samples SRR3473984, SRR3473985, SRR3473984 contain reads covering exon 1. These are Wildtype samples.
- Samples SRR3473987, SRR3473988, SRR3473989 DO NOT contain reads covering exon 1. These are Knockout samples.

</details>
</div>  

### **Read quantification**

**GS- This needs to be updated in accordance with whats in the [runsheet](https://docs.google.com/document/d/1n5k8U3RyBTr5cXcJ34hSmAqBqF7Perow7MOZp0DoZFM/edit#heading=h.vyktmcp6jqez)**  

Following read alignment, we use the alignment files to calculate raw gene-count data for each sample. We can then use these count files (called gene-count matrix), to identify differentially expressed genes. 

To perform read quantification, nf-core/rnaseq uses 

**GS- This needs to be updated in accordance with whats in the [runsheet](https://docs.google.com/document/d/1n5k8U3RyBTr5cXcJ34hSmAqBqF7Perow7MOZp0DoZFM/edit#heading=h.vyktmcp6jqez)**  
<div class="challenge">

### **Challenge**{.unlisted}
- Identify the output folder in which the alignment and quantification results are placed. 
- Identify and check out a few files which you think are important.
<br>**HINT** You can use the unix command `cat` which prints content of a file onto the terminal.

<details>
<summary>Solution</summary>
There are quite a few files generated in this step. We can try and discuss some of them.
</details>
</div>  

<div class="keypoints">

### **Key points**{.unlisted}
- Splice aware alignemnet tools like HISAT2 and STAR must be used when aligning RNA-seq reads to a reference genome.
- ?? **GS- a key point needed for quantification**
</div>  

