---
fig-cap-location: top
---

# **nfcore-rnaseq : Parameters**


<div class="questions">

### **Questions**

- What are the important nfcore-rnaseq parameters
</div>  

<div class="objectives">

### **Objectives**

- Understanding the nfcore-rnaseq parameters
</div>  


### **The nfcore-rnaseq command**

The nfcore-rnaseq pipeline can be run using a single command with deafult parameters.

<font size="3">
```default

  nextflow run $Path_to_nfcore-rnaseq \         # The nfcore-rnaseq excecutable
    --input $Path_to_samplesheet.csv \        # samplesheet file-name
    -profile singularity \                    # For pre-installed softwares
    --fasta $Genome_fasta_file \              # Genome sequence file
    --gtf $Path_to_Genome.gtf \               # gtf - gene locations on genome
    --star_index $Path_to_index file \        # Formatted file for aligner
    --max_memory '6 GB' --max_cpus 2 \        # Memory and cpu resources 
    --outdir $Path_to_results \               # results folder
    -with-report excecution_report.html \     # Excecution log file-name 
    -with-timeline timeline_report.html       # Timeline log file-name

```
</font>


This section introduces the user to the important nfcore-rnaseq parameters


### **Samplesheet sequence data**
- We need a [samplesheet](https://nf-co.re/rnaseq/3.7/usage#samplesheet-input) with information about the samples you would like to analyse.
- If the samplesheet is located in a different folder, you can use this parameter to specify its path. 

![](/fig/elaborate_samplesheet.png)

- The samplesheet is a comma-separated file with 4 columns, and a header row as shown in the example samplesheet above. 
- The pipeline will auto-detect whether a sample is single- or paired-end using the information provided in the samplesheet. 
- The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 4 columns to match those defined in the table below.

![Samplesheet columns](/fig/samplesheet_description.png)



<div class="challenge">

### Challenge
- The samplesheet which we will be using for today's hands-on is already created and available at `~/base_directory/working_directory/samplesheet.csv`.
- Can you open this file using the utility `vi` and figure out what is the difference between ours and the one shown as a example, above?
<details>
<summary>Solution</summary>
The samplesheet above consists of both single- and paired-end data. It is for 6 samples, where TREATMENT_REP3 has been sequenced twice.
The samplesheet to be used in today's analysis is a lot simpler, with only single-end reads. 
</details>
</div>  

### **Other input/output options**
```default
--input   Path to comma-separated file containing information about the samples.
--outdir  The output directory where the results will be saved.
```

#### *Reference genome options*
```default
--genome            Name of iGenomes reference. 
--star_index        Path to directory or tar.gz archive for pre-built STAR index.
--hisat2_index      Path to directory or tar.gz archive for pre-built HISAT2 index.
--save_reference    If generated by the pipeline save the STAR index in the results directory.
```

#### *Alignment options*
```default
--aligner           Alignment algorithm to use.
--pseudo_aligner    Pseudo aligner to use.
```

#### *Process skipping/use-alternate options*
```default
--deseq2_vst      Use vst transformation instead of rlog with DESeq2.
--skip_fastqc     Skip FastQC.
--skip_multiqc    Skip MultiQC.
```

- For details of all parameters in nfcore click [here](https://nf-co.re/rnaseq/3.8.1/parameters).

<div class="keypoints">

### **Key points**
- A single nfcore command can be run for the complete rnaseq pipeline.
- It provides high flexibility change resources such as processing-memory and cpus.  
- The pipeline works with containers such a sigularity and docker.
- nfcore  provides flexibility to use alternative tools (when avaialble) for a particular step and also skip non-madatory steps in the rnaseq pipeline.
</div>  
