<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA-Seq using Nextflow: Day 1 – download_data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../lesson.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RNA-Seq using Nextflow: Day 1</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://sydney-informatics-hub.github.io/training.RNAseq.series-quarto/">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../setup.html">Set-up</a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-1-kick-off" role="button" data-bs-toggle="dropdown" aria-expanded="false">Day 1 kick off</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-1-kick-off">    
        <li>
    <a class="dropdown-item" href="../notebooks/1.0_Learning_objectives.html">
 <span class="dropdown-text">Learning objectives</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://docs.google.com/presentation/d/1Na2nAlK34N196ONzMazGbpxvd-h7Ltx82webeLlN4RU/edit?usp=sharing">
 <span class="dropdown-text">Introductory slides</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.1_Download_data.html">
 <span class="dropdown-text">Prepare your data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.2_Running_nfcore_command.html">
 <span class="dropdown-text">Run the pipeline</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-why-use-nf-corernaseq" role="button" data-bs-toggle="dropdown" aria-expanded="false">Why use nf-core/rnaseq?</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-why-use-nf-corernaseq">    
        <li>
    <a class="dropdown-item" href="../notebooks/2.0_intro_to_nextflow_and_nfcore.html">
 <span class="dropdown-text">Introduction to Nextflow</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.1_nfcore-rnaseq.html">
 <span class="dropdown-text">Introduction to nf-core/rnaseq</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.2_nfcore-rnaseq_otherImportantParameters.html">
 <span class="dropdown-text">Customising nf-core/rnaseq</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-what-is-nf-corernaseq-doing" role="button" data-bs-toggle="dropdown" aria-expanded="false">What is nf-core/rnaseq doing?</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-what-is-nf-corernaseq-doing">    
        <li>
    <a class="dropdown-item" href="../notebooks/3.0_The_pipeline.html">
 <span class="dropdown-text">RNA-seq workflow overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.1_Interpreting_results.html">
 <span class="dropdown-text">nf-core/rnaseq results</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.2_fastQ_and_fastQC.html">
 <span class="dropdown-text">Raw sequencing data QC</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.3_Trimming.html">
 <span class="dropdown-text">Read trimming</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.4_Alignment_using_STAR.html">
 <span class="dropdown-text">Read alignment and quantification</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.5_Interpreting_results_2.html">
 <span class="dropdown-text">Nextflow logs</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../notebooks/4_Summary_Day1.html">Day 1 wrap up</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#prepare-your-data-for-processing" id="toc-prepare-your-data-for-processing" class="nav-link active" data-scroll-target="#prepare-your-data-for-processing"><strong>Prepare your data for processing</strong></a>
  <ul class="collapse">
  <li><a href="#log-in-to-nimbus" id="toc-log-in-to-nimbus" class="nav-link" data-scroll-target="#log-in-to-nimbus"><strong>Log in to Nimbus</strong></a></li>
  <li><a href="#make-your-base-directory" id="toc-make-your-base-directory" class="nav-link" data-scroll-target="#make-your-base-directory"><strong>Make your base directory</strong></a></li>
  <li><a href="#download-the-files-to-be-used-in-this-workshop" id="toc-download-the-files-to-be-used-in-this-workshop" class="nav-link" data-scroll-target="#download-the-files-to-be-used-in-this-workshop"><strong>Download the files to be used in this workshop</strong></a></li>
  <li><a href="#look-at-the-sample-sheet" id="toc-look-at-the-sample-sheet" class="nav-link" data-scroll-target="#look-at-the-sample-sheet"><strong>Look at the sample sheet</strong></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="prepare-your-data-for-processing" class="level1">
<h1><strong>Prepare your data for processing</strong></h1>
<section id="questions" class="level3 unlisted questions">
<h3 class="unlisted anchored" data-anchor-id="questions"><strong>Questions</strong></h3>
<ul>
<li>How do I set up my base directory?</li>
<li>How do I download the required input files?</li>
</ul>
</section>
<p><br> In this workshop series we are working with a real dataset from a study of a knockout mouse model by <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-2801-4">Corley et al.&nbsp;(2016)</a>. Before we can get started, we need to download the materials used in this workshop and organise our working space on the command line.</p>
<div class="callout-note callout callout-style-simple callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Copying the code from the <code>grey</code> boxes on training materials
</div>
</div>
<div class="callout-body-container callout-body">
<p>In this workshop we need to copy code from the grey boxes in the training materials and run it in the terminal. If you hover your mouse over a grey box on the website, a clipboard icon will appear on the right side. Click on the clipboard logo to copy the code. Test it out with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>ssh training@###.###.###.###</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<section id="log-in-to-nimbus" class="level2">
<h2 class="anchored" data-anchor-id="log-in-to-nimbus"><strong>Log in to Nimbus</strong></h2>
<p>Go to the terminal window in which you have logged on to Nimbus. If you have accidently closed it, you will need to repeat steps on the <a href="https://sydney-informatics-hub.github.io/rna-seq-pt1-quarto/setup.html">setup</a> page again. Check where you are by typing:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>pwd</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>It should return <code>/home/training</code>. To get back here at any point, you can always type:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>cd ~</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="make-your-base-directory" class="level2">
<h2 class="anchored" data-anchor-id="make-your-base-directory"><strong>Make your base directory</strong></h2>
<p>This is where we will be working for the next 2 days. In your terminal once you have logged into Nimbus and are sure you’re in <code>/home/training</code>, type the following:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>mkdir base_directory</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="download-the-files-to-be-used-in-this-workshop" class="level2">
<h2 class="anchored" data-anchor-id="download-the-files-to-be-used-in-this-workshop"><strong>Download the files to be used in this workshop</strong></h2>
<p>In your terminal, type the following to move into to the base_directory:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>cd base_directory</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Then type the following to download the input files: <strong>GS- ADD CLOUDSTOR LINK WITH FINAL MATERIALS</strong></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>cloudstor link to add</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The files are packaged up to save space. Let’s unpack it:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>tar -zxvf working_directory.tar.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This creates a directory called <code>working_directory</code>. Move into this directory by running:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>cd working_directory</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>You can list all the files in your current directory by running the list command:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>ls -lh</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>It should return this:</p>
<p><font size="3"></font></p><font size="3">
<div class="sourceCode" id="cb10"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>-rw-------@ 1 training  training   919K 22 Apr 15:11 FULL_count_matrix.txt</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>-rw-------@ 1 training  training   1.3K 30 Aug 11:53 README</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>-rwxr-xr-x@ 1 training  training   3.4M 28 Jul 06:11 SRR3473984_selected.fastq.gz</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>-rwxr-xr-x@ 1 training  training   3.1M 28 Jul 06:11 SRR3473985_selected.fastq.gz</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>-rwxr-xr-x@ 1 training  training   3.0M 28 Jul 06:11 SRR3473986_selected.fastq.gz</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>-rwxr-xr-x@ 1 training  training   3.3M 28 Jul 06:11 SRR3473987_selected.fastq.gz</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>-rwxr-xr-x@ 1 training  training   4.2M 28 Jul 06:11 SRR3473988_selected.fastq.gz</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>-rwxr-xr-x@ 1 training  training   3.0M 28 Jul 06:11 SRR3473989_selected.fastq.gz</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>-rw-------@ 1 training  training   4.8K 28 Jul 06:11 nextflow.config</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>-rw-r--r--@ 1 training  training    18K 30 Aug 11:33 rnaseq_DE_analysis_Day2.Rmd</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>-rw-r--r--@ 1 training  training   5.0M 30 Aug 11:37 rnaseq_DE_analysis_Day2.html</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>-rw-------@ 1 training  training   330B 28 Jul 07:25 samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</font><p><font size="3"></font></p>
<p>The <code>working_directory</code> folder contains the following files:</p>
<ul>
<li>6 gzipped fastq files: <code>SRR*_selected.fastq.gz</code></li>
<li>A samplesheet file required by the nf-core/rnaseq pipeline: <code>samplesheet.csv</code></li>
<li>The nf-core/rnaseq pipeline configuration file: <code>nextflow.config</code><br>
</li>
<li>The genome-wide gene count matrix file for day 2: <code>FULL_count_matrix.txt</code><br>
</li>
<li>The lessons and code for day 2: <code>rnaseq_DE_analysis_Day2.Rmd</code> and <code>rnaseq_DE_analysis_Day2.html</code></li>
</ul>
</section>
<section id="look-at-the-sample-sheet" class="level2">
<h2 class="anchored" data-anchor-id="look-at-the-sample-sheet"><strong>Look at the sample sheet</strong></h2>
<p>A sample sheet is a standard input rquirement of nf-core pipelines, including nf-core/rnaseq. This sheet contains information on each sample being processed with this pipeline. Take a look at the <a href="https://nf-co.re/rnaseq/3.7/usage#samplesheet-input">nf-core/rnaseq documentation</a> for more information. For example:</p>
<p><img src="../fig/elaborate_samplesheet.png" class="img-fluid"></p>
<p>The sample sheet we are using today contains a <strong>header row</strong> and <strong>one row per sample</strong> being processed. It has 4 columns, each separated by the comma. The pipeline will auto-detect whether a sample is single- or paired-end using the information provided in the samplesheet. The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 4 columns to match those defined in the table below:</p>
<p><img src="../fig/samplesheet_description.png" class="img-fluid"></p>
<section id="challenge" class="level3 unlisted challenge">
<h3 class="unlisted anchored" data-anchor-id="challenge">Challenge</h3>
<p>Can you view the provided sample sheet using the <code>cat</code> command and identify differences between our sheet the one shown in the example screenshot above?</p>
<details>
<summary>
Solution
</summary>
<p>View samplesheet.csv by running:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>cat ~/base_directory/working_directory/samplesheet.csv </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Our sample sheet looks like this: <img src="../fig/our_samplesheet_final.png" class="img-fluid" style="width:70.0%" alt="Today’s samplesheet"></p>
<p>Both sample sheets contain 6 samples. However, the example samplesheet consists of both single- and paired-end data for the control samples and TREATMENT_REP3 has been sequenced twice.</p>
<p>Our sample sheet is a lot simpler, with only single-end reads. Note that the column for the reverse-reads is empty in our sample sheet.</p>
</details>
</section>
<p>Now that we have prepared our work space and downloaded the required files, we will proceed to running the nf-core/rnseq pipeline. Before proceeding:</p>
<p>Check your path by running:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>pwd</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This should return: <code>/home/training/base_directory/working_directory</code>. If you are not here, move into the above path by running:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>cd /home/training/base_directory/working_directory</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><br> Keep your terminal window open and <a href="https://sydney-informatics-hub.github.io/rna-seq-pt1-quarto/notebooks/1.2_Running_nfcore_command.html">proceed to the next lesson</a> by clicking on <strong>Day 1 kick off &gt; Run the pipeline</strong> on the menu bar.</p>
<section id="key-points" class="level3 unlisted keypoints">
<h3 class="unlisted anchored" data-anchor-id="key-points"><strong>Key points</strong></h3>
<ul>
<li>Organise your working space on the command line using directories</li>
<li>The nf-core/rnaseq pipeline requires raw sequence data in the fastq format</li>
<li>We can tell nf-core/rnaseq which fastq files belong to which sample using the sample sheet</li>
</ul>
</section>


</section>
</section>

<p>All materials copyright Sydney Informatics Hub, University of Sydney</p></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>