<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA-Seq using Nextflow: Day 1 – alignment_using_star</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../lesson.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RNA-Seq using Nextflow: Day 1</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../setup.html">Setup</a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-start-day-1" role="button" data-bs-toggle="dropdown" aria-expanded="false">Start Day 1</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-start-day-1">    
        <li>
    <a class="dropdown-item" href="../notebooks/1.0_Learning_objectives.html">
 <span class="dropdown-text">Learning objectives: Day 1</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://unisyd.sharepoint.com/:p:/r/teams/SydneyInformaticsHub2/_layouts/15/Doc.aspx?sourcedoc=%7B02017CE8-31E0-4693-AAA7-27968E003127%7D&amp;file=Sept2022_Introductory_slides_biocommons_template.pptx&amp;action=edit&amp;mobileredirect=true&amp;cid=e9271245-0780-4b25-8982-fb6da8d78cca">
 <span class="dropdown-text">Introductory slides</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.1_Download_data.html">
 <span class="dropdown-text">Download all files</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/Running_nfcore_command.html">
 <span class="dropdown-text">Run nf-core/rnaseq pipeline on Pawsey’s Nimbus Cloud</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-why-use-nf-core-workflows" role="button" data-bs-toggle="dropdown" aria-expanded="false">Why use nf-core workflows?</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-why-use-nf-core-workflows">    
        <li>
    <a class="dropdown-item" href="../notebooks/3_intro_to_nextflow_and_nfcore.html">
 <span class="dropdown-text">Why use Nextflow?</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/4_nfcore-rnaseq.html">
 <span class="dropdown-text">Introduction to nf-core/rnaseq</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/6.1_nfcore-rnaseq_otherImportantParameters.html">
 <span class="dropdown-text">nf-core/rnaseq : Parameters</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-rna-seq-differential-expression-(de-analysis" role="button" data-bs-toggle="dropdown" aria-expanded="false">RNA-seq differential expression (DE) analysis</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-rna-seq-differential-expression-(de-analysis">    
        <li>
    <a class="dropdown-item" href="../notebooks/6.3_Interpreting_results.html">
 <span class="dropdown-text">Results from nf-core/rnaseq</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.1_The_pipeline.html">
 <span class="dropdown-text">The main steps involved in RNA-seq (DE) analysis</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.2_fastQ_and_fastQC.html">
 <span class="dropdown-text">Get raw sequencing data and check quality</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.3_Trimming.html">
 <span class="dropdown-text">Trimming</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.4_Alignment_using_STAR.html">
 <span class="dropdown-text">Alignment to reference genome and quantification</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/6.4_Interpreting_results_2.html">
 <span class="dropdown-text">Helpful logs</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-summary-day-1" role="button" data-bs-toggle="dropdown" aria-expanded="false">Summary Day 1</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-summary-day-1">    
        <li>
    <a class="dropdown-item" href="../notebooks/7_Summary_Day1.html">
 <span class="dropdown-text">Raw sequence to gene-counts</span></a>
  </li>  
    </ul>
  </li>
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#alignment-to-reference-genome-and-quantification" id="toc-alignment-to-reference-genome-and-quantification" class="nav-link active" data-scroll-target="#alignment-to-reference-genome-and-quantification"><strong>Alignment to reference genome and quantification</strong></a>
  <ul class="collapse">
  <li><a href="#questions" id="toc-questions" class="nav-link" data-scroll-target="#questions">Questions</a></li>
  <li><a href="#objectives" id="toc-objectives" class="nav-link" data-scroll-target="#objectives">Objectives</a></li>
  <li><a href="#alignment-to-reference-genome" id="toc-alignment-to-reference-genome" class="nav-link" data-scroll-target="#alignment-to-reference-genome"><strong>Alignment to reference genome</strong></a></li>
  <li><a href="#challenge" id="toc-challenge" class="nav-link" data-scroll-target="#challenge"><strong>Challenge</strong></a></li>
  <li><a href="#pseudo-aligners" id="toc-pseudo-aligners" class="nav-link" data-scroll-target="#pseudo-aligners">Pseudo-aligners</a></li>
  <li><a href="#read-quantification" id="toc-read-quantification" class="nav-link" data-scroll-target="#read-quantification"><strong>Read quantification</strong></a></li>
  <li><a href="#challenge-1" id="toc-challenge-1" class="nav-link" data-scroll-target="#challenge-1"><strong>Challenge</strong></a></li>
  <li><a href="#key-points" id="toc-key-points" class="nav-link" data-scroll-target="#key-points"><strong>Key points</strong></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="alignment-to-reference-genome-and-quantification" class="level1">
<h1><strong>Alignment to reference genome and quantification</strong></h1>
<section id="questions" class="level3 questions">
<h3 class="anchored" data-anchor-id="questions">Questions</h3>
<ul>
<li>How to map sequencing reads to a reference genome?</li>
<li>How to use the mapped alignments to quantify gene-counts</li>
</ul>
</section>
<section id="objectives" class="level3 objectives">
<h3 class="anchored" data-anchor-id="objectives">Objectives</h3>
<ul>
<li>Understand the alignment/mapping process</li>
<li>Understand the output sam/bam formats</li>
<li>Understand the convertion of read-alignments to gene-counts</li>
</ul>
</section>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p></p><figcaption class="figure-caption"><strong>Stage 2 of <code>nf-core/rnaseq</code>: (Genome alignment and quantification):</strong></figcaption><p></p>
<p><img src="../fig/nfcore_stage2And3.png" class="img-fluid figure-img"></p>
</figure>
</div>
<section id="alignment-to-reference-genome" class="level3">
<h3 class="anchored" data-anchor-id="alignment-to-reference-genome"><strong>Alignment to reference genome</strong></h3>
<ul>
<li>After trimming, the next step of processing RNA-seq data is to map the reads to a reference genome. RNA-seq specific aligner needs to be used to accomodate splicing. The most commonly used aligners for short-read sequencing are <a href="https://github.com/alexdobin/STAR">STAR</a>, <a href="http://daehwankimlab.github.io/hisat2/">HISAT2</a> etc.</li>
<li>The output of the alignment step is a <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">bam</a> file, which is a compressed (“B”inary) form of a <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">sam</a> file.</li>
</ul>
<p><br>The next challenge will be done in breakout rooms. A facilitator will handle the big bam files on their side using a tool call <code>IGV</code> which will be followed by a discussion.</p>
</section>
<section id="challenge" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge"><strong>Challenge</strong></h3>
<ul>
<li>Open the application: The Integrative Genomics Viewer <a href="https://software.broadinstitute.org/software/igv/">(IGV)</a>.</li>
<li>Open the 6 pre-aligned bam files. These files contain the reads which are aligned specifically to the region flanking our gene of interest which is <code>Gtf2ird1</code>.</li>
<li>This will be a good time to revisit the <code>The case study</code> which is mentioned in the <a href="https://unisyd.sharepoint.com/:p:/r/teams/SydneyInformaticsHub2/_layouts/15/Doc.aspx?sourcedoc=%7B02017CE8-31E0-4693-AAA7-27968E003127%7D&amp;file=Sept2022_Introductory_slides_biocommons_template.pptx&amp;action=edit&amp;mobileredirect=true&amp;cid=8c296cbc-3005-48d3-b977-bef7c517961e">Introductory slides</a>.</li>
<li>Once the bam files are opened in the IGV viewer, we can navigate to the exact region of the gene <code>Gtf2ird1</code>.</li>
<li><strong>We have forgotten to label our samples</strong> and don’t know which samples belong to the wildtype or knockout group! So, instead, we will use the alignments and assign samples to their correct treatment group (wildtype or knockout)</li>
</ul>
<p><br> <img src="../fig/Gtf2ird1_exon1.png" class="img-fluid"></p>
<ul>
<li>Based on the observation (which is…), can you identify which samples belong to the <code>WT</code> and which of them are the <code>KO</code> samples? Please paste your answers in the breakout chat window.</li>
</ul>
<details>
<summary>
Solution
</summary>
<ul>
<li>A loss of function mutation of&nbsp;Gtf2ird1&nbsp;was generated by a random insertion of a&nbsp;Myc transgene into the region, resulting in a 40&nbsp;kb deletion surrounding exon 1.</li>
<li>Samples <code>SRR3473984,SRR3473985,SRR3473984</code> contain reads supporting exon 1. These are <code>WT</code> samples.</li>
<li>Samples <code>SRR3473987,SRR3473988,SRR3473989</code> DO NOT contain reads supporting exon 1. These are <code>KO</code> samples.</li>
</ul>
</details>
</section>
<section id="pseudo-aligners" class="level3">
<h3 class="anchored" data-anchor-id="pseudo-aligners">Pseudo-aligners</h3>
<p>As salmon is part of the alignment tool STAR_salmon in nf-core/rnaseq, I will quickly discuss Pseudo-aligners and salmon here</p>
</section>
<section id="read-quantification" class="level3">
<h3 class="anchored" data-anchor-id="read-quantification"><strong>Read quantification</strong></h3>
<p>The reads mapped to the reference genome are then converted to a matrix which contains individual gene-counts per sample using the tool <a href="http://subread.sourceforge.net/">featurecounts</a> . The gene-count matrix can then be used for identification of differentially expressed genes.</p>
</section>
<section id="challenge-1" class="level3 challenge">
<h3 class="anchored" data-anchor-id="challenge-1"><strong>Challenge</strong></h3>
<ul>
<li>Identify the output folder in which the alignment and quantification results are placed.</li>
<li>Identify and check ouit a few files <strong>HINT</strong> You can use the unix command <code>cat</code> which prints content of a file onto the terminal.</li>
</ul>
<details>
<summary>
Solution
</summary>
There are quite a few files geenrated in this step. We can try and discuss a few which are of interest.
</details>
</section>
<section id="key-points" class="level3 keypoints">
<h3 class="anchored" data-anchor-id="key-points"><strong>Key points</strong></h3>
<ul>
<li>Tools such as STAR and HISAT2 can be used for splice-aware alignemnet of RNA-Seq reads.</li>
<li>Tools such as featurecounts can be used to quantify the aligned-reads to gene-counts.</li>
</ul>
</section>


</section>

<p>All materials copyright Sydney Informatics Hub, University of Sydney</p></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>