<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA-Seq using Nextflow: Day 1 – interpreting_results</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../lesson.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RNA-Seq using Nextflow: Day 1</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://sydney-informatics-hub.github.io/training.RNAseq.series-quarto/">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../notebooks/0.0_about.html">About</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../setup.html">Set-up</a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-1-kick-off" role="button" data-bs-toggle="dropdown" aria-expanded="false">Day 1 kick off</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-1-kick-off">    
        <li>
    <a class="dropdown-item" href="../notebooks/1.0_Learning_objectives.html">
 <span class="dropdown-text">Learning objectives</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://docs.google.com/presentation/d/1Na2nAlK34N196ONzMazGbpxvd-h7Ltx82webeLlN4RU/edit?usp=sharing">
 <span class="dropdown-text">Introductory slides</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.1_Download_data.html">
 <span class="dropdown-text">Prepare your data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.2_Running_nfcore_command.html">
 <span class="dropdown-text">Run the pipeline</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-why-use-nf-corernaseq" role="button" data-bs-toggle="dropdown" aria-expanded="false">Why use nf-core/rnaseq?</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-why-use-nf-corernaseq">    
        <li>
    <a class="dropdown-item" href="../notebooks/2.0_intro_to_nextflow_and_nfcore.html">
 <span class="dropdown-text">Introduction to Nextflow</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.1_nfcore-rnaseq.html">
 <span class="dropdown-text">Introduction to nf-core/rnaseq</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.2_nfcore-rnaseq_otherImportantParameters.html">
 <span class="dropdown-text">Customising nf-core/rnaseq</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-what-is-nf-corernaseq-doing" role="button" data-bs-toggle="dropdown" aria-expanded="false">What is nf-core/rnaseq doing?</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-what-is-nf-corernaseq-doing">    
        <li>
    <a class="dropdown-item" href="../notebooks/3.0_The_pipeline.html">
 <span class="dropdown-text">RNA-seq workflow overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.1_Interpreting_results.html">
 <span class="dropdown-text">nf-core/rnaseq results</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.2_fastQ_and_fastQC.html">
 <span class="dropdown-text">Raw sequencing data QC</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.3_Trimming.html">
 <span class="dropdown-text">Read trimming</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.4_Alignment_using_STAR.html">
 <span class="dropdown-text">Read alignment and quantification</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.5_Interpreting_results_2.html">
 <span class="dropdown-text">Nextflow logs</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../notebooks/4_Summary_Day1.html">Day 1 wrap up</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#nf-corernaseq-results" id="toc-nf-corernaseq-results" class="nav-link active" data-scroll-target="#nf-corernaseq-results"><strong>nf-core/rnaseq results</strong></a>
  <ul class="collapse">
  <li><a href="#the-results-directory" id="toc-the-results-directory" class="nav-link" data-scroll-target="#the-results-directory"><strong>The results directory</strong></a></li>
  <li><a href="#copy-the-results-folder-to-your-local-machine" id="toc-copy-the-results-folder-to-your-local-machine" class="nav-link" data-scroll-target="#copy-the-results-folder-to-your-local-machine"><strong>Copy the results folder to your local machine</strong></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="nf-corernaseq-results" class="level1">
<h1><strong>nf-core/rnaseq results</strong></h1>
<section id="questions" class="level3 unlisted questions">
<h3 class="unlisted anchored" data-anchor-id="questions"><strong>Questions</strong></h3>
<ul>
<li>How are the results from the nf-core/rnaseq organised?</li>
<li>How can I transfer the results from Nimbus to my local computer?</li>
</ul>
</section>
<p><br> Lets start by checking that the nf-core/rnaseq command has completed. Bring up the terminal window you left your command running in. If your run was successful, your terminal should display a Pipeline completed successfully message, same as the image below:</p>
<p><img src="../fig/completed_nfcore_rnaseq_run.png" class="img-fluid"></p>
<p>The pipeline completion message tells us some useful information including when the run finished, how long it took to run, and how many CPU hours it consumed. If you scroll up in your terminal, you’ll see just how many processes were run, and can confirm they all ran successfully.</p>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
<strong>A comment on running time</strong>
</div>
</div>
<div class="callout-body-container callout-body">
<p>You may notice the walltime of your command was different to the screenshot above. This command was run on a different Nimbus machine to those we are working on today.</p>
<p><strong>Can you think of any differences between computers and the way the command can be run that might speed up the running time of the pipeline?</strong></p>
</div>
</div>
<section id="the-results-directory" class="level3">
<h3 class="anchored" data-anchor-id="the-results-directory"><strong>The results directory</strong></h3>
<p>Lets take a look at the results output by the pipeline. Remember that we told the nf-core/rnaseq run command to output all results to a directory named ‘results’ using the <code>--outdir</code> flag.</p>
<p>To view the contents of the results folder in your terminal, run:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>ls -lh results</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>It should return a list of directories like this:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>drwxrwxr-x  2 training training 4.0K Sep 19 03:35 fastqc</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>drwxrwxr-x  3 training training 4.0K Sep 19 03:55 multiqc</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>drwxrwxr-x  2 training training 4.0K Sep 19 03:56 pipeline_info</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>drwxrwxr-x 19 training training 4.0K Sep 19 03:55 star_salmon</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>drwxrwxr-x  3 training training 4.0K Sep 19 03:35 trimgalore</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Each of these directories contains results and files generated by various stages of the pipeline for each sample. We’re going to go through these directories as we talk through the stages run by the pipeline.</p>
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
MultiQC output logs
</div>
</div>
<div class="callout-body-container callout-body">
<p>One thing we won’t have time to discuss today is the MultiQC reports. <a href="https://multiqc.info/">MultiQC</a> is a great tool for reporting summary metrics for many different bioinformatic tools.</p>
<p>In our nf-core/rnaseq run, MultiQC has searched the <code>results</code> directory for analysis logs and compiles an .html report for each tool.</p>
</div>
</div>
</section>
<section id="copy-the-results-folder-to-your-local-machine" class="level3">
<h3 class="anchored" data-anchor-id="copy-the-results-folder-to-your-local-machine"><strong>Copy the results folder to your local machine</strong></h3>
<p>Some of the files the pipeline generated are in <code>.html</code> format which cannot view in the terminal so we will copy them to our local computer and use our web browsers to view them.</p>
<p>Transferring the files from Nimbus to your local computer can be done using the scp command. If you experience any issues, let a facilitator know in the Slack channel. We can supply participants with a link to a prepared results folder.</p>
<p>To transfer the results files:</p>
<ol type="1">
<li><p>Open a new terminal window. <strong>Do not</strong> log in to Nimbus.</p></li>
<li><p>Navigate to the path of your choice on your <strong>local computer</strong> by running:</p></li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>cd /path/to/directory</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ol start="3" type="1">
<li>Make a folder to save your results in:</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>mkdir RNAseq_workshop</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ol start="4" type="1">
<li>Download the fastqc and trimgalore results from Nimbus to your local computer:</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>scp -r training@###.###.###.###:~/base_directory/working_directory/results/fastqc ./RNAseq_workshop/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>scp -r training@###.###.###.###:~/base_directory/working_directory/results/trimgalore ./RNAseq_workshop/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="callout-warning callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Warning for MobaXTerm users
</div>
</div>
<div class="callout-body-container callout-body">
<p>Accessing your local files using a terminal application on Windows machines can be a bit complicated. When downloading results from Nimbus using MobaXterm, you can try the following:</p>
<ol type="1">
<li>Check where your default home directory is by running <code>pwd</code>. It should return: <code>/home/mobaxterm</code></li>
<li>See what folders you can access from there using <code>ls</code></li>
<li>Move into your Desktop folder using <code>cd Desktop</code> and then run the <code>mkdir</code> and <code>scp</code> commands above.</li>
</ol>
<p><img src="../fig/scp_mobaxterm_example.png" class="img-fluid" style="width:120.0%"></p>
</div>
</div>
<p>We can now use a web browser on our computers to open the html reports. We’re going to investigate the files and results produced by the pipeline as we step through the workflow run by the nf-core/rnaseq pipeline, starting with the raw sequence data quality checks.</p>
<p><a href="https://sydney-informatics-hub.github.io/rna-seq-pt1-quarto/notebooks/3.2_fastQ_and_fastQC.html">Proceed to the next lesson</a> by clicking on <strong>What is nf-core/rnaseq doing? &gt; Raw sequencing data QC</strong> on the menu bar.</p>
</section>
<section id="key-points" class="level3 unlisted keypoints">
<h3 class="unlisted anchored" data-anchor-id="key-points"><strong>Key points</strong></h3>
<ul>
<li>The nf-core/rnaseq pipeline generates independant sub-folders in the main results folder for individual processes run by the pipeline.</li>
<li>The name of the results folder can be customised using the <code>--outdir</code> flag.</li>
<li>Some of the results files like quality control reports from FastQC and MultiQC are in html format and can’t be viewed using your terminal.</li>
</ul>
</section>


</section>

<p>All materials copyright Sydney Informatics Hub, University of Sydney</p></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>