<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>RNA-Seq using Nextflow: Day 1 – fastq_and_fastqc</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../lesson.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RNA-Seq using Nextflow: Day 1</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://sydney-informatics-hub.github.io/training.RNAseq.series-quarto/">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../setup.html">Set-up</a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-day-1-kick-off" role="button" data-bs-toggle="dropdown" aria-expanded="false">Day 1 kick off</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-day-1-kick-off">    
        <li>
    <a class="dropdown-item" href="../notebooks/1.0_Learning_objectives.html">
 <span class="dropdown-text">Learning objectives</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://docs.google.com/presentation/d/1Na2nAlK34N196ONzMazGbpxvd-h7Ltx82webeLlN4RU/edit?usp=sharing">
 <span class="dropdown-text">Introductory slides</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.1_Download_data.html">
 <span class="dropdown-text">Prepare your data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/1.2_Running_nfcore_command.html">
 <span class="dropdown-text">Run the pipeline</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-why-use-nf-corernaseq" role="button" data-bs-toggle="dropdown" aria-expanded="false">Why use nf-core/rnaseq?</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-why-use-nf-corernaseq">    
        <li>
    <a class="dropdown-item" href="../notebooks/2.0_intro_to_nextflow_and_nfcore.html">
 <span class="dropdown-text">Introduction to Nextflow</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.1_nfcore-rnaseq.html">
 <span class="dropdown-text">Introduction to nf-core/rnaseq</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/2.2_nfcore-rnaseq_otherImportantParameters.html">
 <span class="dropdown-text">Customising nf-core/rnaseq</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-what-is-nf-corernaseq-doing" role="button" data-bs-toggle="dropdown" aria-expanded="false">What is nf-core/rnaseq doing?</a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-what-is-nf-corernaseq-doing">    
        <li>
    <a class="dropdown-item" href="../notebooks/3.0_The_pipeline.html">
 <span class="dropdown-text">RNA-seq workflow overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.1_Interpreting_results.html">
 <span class="dropdown-text">nf-core/rnaseq results</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.2_fastQ_and_fastQC.html">
 <span class="dropdown-text">Raw sequencing data QC</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.3_Trimming.html">
 <span class="dropdown-text">Read trimming</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.4_Alignment_using_STAR.html">
 <span class="dropdown-text">Read alignment and quantification</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../notebooks/3.5_Interpreting_results_2.html">
 <span class="dropdown-text">Nextflow logs</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../notebooks/4_Summary_Day1.html">Day 1 wrap up</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#raw-sequence-data-quality-control" id="toc-raw-sequence-data-quality-control" class="nav-link active" data-scroll-target="#raw-sequence-data-quality-control"><strong>Raw sequence data quality control</strong></a>
  <ul class="collapse">
  <li><a href="#fastq-format" id="toc-fastq-format" class="nav-link" data-scroll-target="#fastq-format"><strong>Fastq format</strong></a></li>
  <li><a href="#what-are-phred-quality-scores" id="toc-what-are-phred-quality-scores" class="nav-link" data-scroll-target="#what-are-phred-quality-scores"><strong>What are Phred quality scores?</strong></a></li>
  <li><a href="#what-quality-metrics-can-we-evaluate-with-fastqc" id="toc-what-quality-metrics-can-we-evaluate-with-fastqc" class="nav-link" data-scroll-target="#what-quality-metrics-can-we-evaluate-with-fastqc"><strong>What quality metrics can we evaluate with FastQC?</strong></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">



<section id="raw-sequence-data-quality-control" class="level1">
<h1><strong>Raw sequence data quality control</strong></h1>
<section id="questions" class="level3 unlisted questions">
<h3 class="unlisted anchored" data-anchor-id="questions"><strong>Questions</strong></h3>
<ul>
<li>What is the Fastq format?</li>
<li>What are Phred quality scores?</li>
<li>How can I evaluate the quality of my raw data using FastQC?</li>
</ul>
</section>
<p><br> We’re starting at the first stage of the nf-core/rnaseq workflow, specifically with raw sequence data (red box below). When your data is sequenced, it will be output in the fastq format by the sequencing machine.</p>
<p><img src="../fig/nfcore_stage1.1.png" class="img-fluid"></p>
<section id="fastq-format" class="level2">
<h2 class="anchored" data-anchor-id="fastq-format"><strong>Fastq format</strong></h2>
<p><a href="https://en.wikipedia.org/wiki/FASTQ_format">Fastq</a> is a text-based format for storing both a biological sequence (usually nucleotide sequence) and it’s corresponding quality score. Each entry in a fastq file will consist of 4 lines:</p>
<ol type="1">
<li>A sequence identifier (label)</li>
<li>The nucleotide sequence</li>
<li>A separator line, usually just a plus (+) sign</li>
<li>The base call quality score per nucleotide. These are Phred scored, using ASCII characters.</li>
</ol>
<p><img src="../fig/fastq_format.jpg" class="img-fluid"></p>
</section>
<section id="what-are-phred-quality-scores" class="level2">
<h2 class="anchored" data-anchor-id="what-are-phred-quality-scores"><strong>What are Phred quality scores?</strong></h2>
<p>Phred quality scores are used to indicate the quality of a base call. The Phred score corresponds to the probability that the base was called correctly. Take a look at <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531872-Phred-scaled-quality-scores">GATK’s explanation of Phred scores</a> for more information.</p>
<p><img src="../fig/Phred_short.png" class="img-fluid"></p>
</section>
<section id="what-quality-metrics-can-we-evaluate-with-fastqc" class="level2">
<h2 class="anchored" data-anchor-id="what-quality-metrics-can-we-evaluate-with-fastqc"><strong>What quality metrics can we evaluate with FastQC?</strong></h2>
<p><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> is a popular tool used to evaluate fastq files. It provides a set of metrics which we can use to get a sense of whether our raw data has any problems that might impact our ability to analyse it downstream. For example, the image below is an example of the per base sequence quality plot that FastQC generates. In this example, we’re looking at a dataset consisting of 101 bp reads. FastQC has looked at the Phred scores of all nucleotides across all reads and plot the distribution of Phred scores for each nucleotide position.</p>
<p>Open up the results folder you downloaded to your computer and open the fastqc subdirectory. Take a look at the files ending in fastqc.html by opening them in a web browser and answering the challenge questions below:</p>
<p><img src="../fig/fastqQC.png" class="img-fluid"></p>
<section id="challenge-1" class="level3 unlisted challenge">
<h3 class="unlisted anchored" data-anchor-id="challenge-1"><strong>Challenge #1</strong></h3>
<ol type="1">
<li>How many sequences were in SRR3473989.fastq?</li>
<li>How long are the sequence reads in SRR3473989.fastq?</li>
</ol>
<details>
<summary>
Solution
</summary>
<ol type="1">
<li>There are <code>59887</code> sequences in the file SRR3473989.fastq.</li>
<li>The reads in the file SRR3473989.fastq are of length <code>101 bp</code>.</li>
</ol>
</details>
</section>
<section id="challenge-2" class="level3 unlisted challenge">
<h3 class="unlisted anchored" data-anchor-id="challenge-2"><strong>Challenge #2</strong></h3>
<p>Looking at the <strong>Per base sequence quality plot</strong> for SRR3473989.fastq, answer the following questions.</p>
<ol type="1">
<li>Which part of the reads tend to have worse per base sequence quality?</li>
<li>Do you think this dataset is of ‘good’ quality? Why or why not?</li>
<li>Any suggestions to improve the quality of our raw reads?</li>
</ol>
<details>
<summary>
Solution
</summary>
<ol type="1">
<li>Reads which tend to have worse per base sequence quality are towards the right hand side (3’ end).</li>
<li>The color coding separates out regions of good quality (Red PhredQ &gt; 28) from the rest. Overall yes, as most of the regions of the reads show quality values in red.</li>
<li>We can trim the bases towards the 3’-end and hope to improve the overall read-quality. But trimming by quality for RNA-seq data has its pros and cons.</li>
</ol>
</details>
</section>
<p><a href="https://sydney-informatics-hub.github.io/rna-seq-pt1-quarto/notebooks/3.3_Trimming.html">Proceed to the next lesson</a> by clicking on <strong>What is nf-core/rnaseq doing? &gt; Read trimming</strong> on the menu bar.</p>
<section id="key-points" class="level3 unlisted keypoints">
<h3 class="unlisted anchored" data-anchor-id="key-points">Key points</h3>
<ul>
<li>Fastq is a standard format for sequencing reads.</li>
<li>Phred Quality Scores are used to measure base qualities of sequencing reads.</li>
<li>FastQC can be used for evaluation.</li>
</ul>
</section>


</section>
</section>

<p>All materials copyright Sydney Informatics Hub, University of Sydney</p></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>